src/App.tsx

```diff
@@ -1,14 +1,35 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { useAppDispatch } from './app/hooks';
+import { getTodos } from './api';
+import { actions as todosActions } from './features/todos';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(todosFromServer => {
+        dispatch(todosActions.setTodos(todosFromServer));
+      })
+      .catch(error => {
+        throw new Error('Error fetching todos', error);
```

Comment:
do not throw an error here - you handled it already (or should handle it - change some state value to show the error message, etc.), so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)

FYI: take into account in the future